src/js/components/RangeInput/StyledRangeInput.js

```diff
@@ -1,14 +1,75 @@
+import { rgba } from 'polished';
 import styled, { css } from 'styled-components';
 
 import { focusStyle, normalizeColor, parseMetricToNum } from '../../utils';
 import { defaultProps } from '../../default-props';
 
+// opacity of the bound trumps the track opacity
+const getBoundOpacity = (props, bound) => {
+  return props.theme.rangeInput &&
+    props.theme.rangeInput.track &&
+    props.theme.rangeInput.track[bound] &&
+    props.theme.rangeInput.track[bound].opacity
+    ? props.theme.rangeInput.track[bound].opacity
+    : 1;
+};
+
+const getBoundColor = (props, bound) => {
+  if (
+    props.theme.rangeInput &&
+    props.theme.rangeInput.track &&
+    props.theme.rangeInput.track[bound] &&
+    props.theme.rangeInput.track[bound].color
+  ) {
+    return rgba(
+      normalizeColor(props.theme.rangeInput.track[bound].color, props.theme),
+      getBoundOpacity(props, bound),
+    );
+  }
+  // If bound color is undefined pick the default track color with bound opacity
+  return rgba(
+    normalizeColor(props.theme.rangeInput.track.color, props.theme),
+    getBoundOpacity(props, bound),
+  );
+};
+
+const trackColorStyle = props => {
+  // backward compatibility in case no bounds are defined
+  if (
+    props.theme.rangeInput &&
+    props.theme.rangeInput.track &&
+    !props.theme.rangeInput.track.lower &&
+    !props.theme.rangeInput.track.upper
+  ) {
+    const color = rgba(
+      normalizeColor(props.theme.rangeInput.track.color, props.theme),
+      0.2,
+    );
+    if (color === 'rgba(0,0,0,0.2)') return `background: ${color}`;
```

Comment:
Why do we need to check for a specific value here? We already know there aren't upper and lower bounds to fall through to.

Reply:
Since we changed the color on base.js from border with opacity to just `border` this condition is used to make sure we are applying the opacity correctly in case the color is indeed the `border` color.
Your comment could have been true if we defaulted the opacity to 0.2, but see my first comment above regarding the concern of adding a default opacity in base.js.

Reply:
Can we add a comment explaining this? It seems a bit obscure otherwise.

Reply:
Done